### PR TITLE
v1.2 Display Sales Report

### PR DESCRIPTION
**Pull request by HyperionNKJ**
Command: display-sales DATE
Output: The sales report window will pop-up. The window consists of a TableView (each row representing a SalesRecord) and a Label (representing the total revenue earned in that day).

Test code is now completed.

### CODE REVIEW

**Comment by HyperionNKJ** [comment]
PR is ready for review.

File: src/main/java/seedu/address/logic/commands/sales/DisplaySalesCommand.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import seedu.address.model.salesrecord.SalesReport;
+
+/**
+ * Display sales report of a specific date
+ */
+public class DisplaySalesCommand extends Command {
+    public static final String COMMAND_WORD = "display-sales";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Generate and display the sales report of a specified "
+            + "date.\n"
+            + "Parameters: DATE (must be in DD-MM-YYYY format) "
```
[the remark this answers is not in the document]
Comment (HyperionNKJ) [reply src/main/java/seedu/address/logic/commands/sales/DisplaySalesCommand.java]: Thanks for the advice. Will take that into consideration for v1.4 😄 If I do use Natty, we can extend its functionalities to other component that uses Date as well.

File: src/main/java/seedu/address/model/salesrecord/SalesReport.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    @Override
+    public int hashCode() {
+        // use this method for custom fields hashing instead of implementing your own
+        return Objects.hash(date, records, totalRevenue);
+    }
+
+    @Override
+    public String toString() {
+        return "Sales Report for  " + getDate();
```
[the remark this answers is not in the document]
Comment (HyperionNKJ) [reply src/main/java/seedu/address/model/salesrecord/SalesReport.java]: Yup, I intended to keep it as concise as possible!

[the remark this answers is not in the document]
Comment (AZhiKai) [reply src/main/java/seedu/address/model/salesrecord/SalesReport.java]: 👍

File: src/main/java/seedu/address/logic/commands/sales/DisplaySalesCommand.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (AZhiKai) [reply src/main/java/seedu/address/logic/commands/sales/DisplaySalesCommand.java]: Yes, that's for sure. 😄 
If you want to consider, perhaps do in 1.3 as 1.4 might be too tight.

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 3]
```diff
@@ -122,6 +124,9 @@ public Command parseCommand(String userInput) throws ParseException {
         case RecordSalesCommand.COMMAND_WORD:
             return new RecordSalesCommandParser().parse(arguments);
```
[the remark this answers is not in the document]
Comment (HyperionNKJ) [reply src/main/java/seedu/address/logic/parser/AddressBookParser.java]: Okay, will implement them soon!

**Comment by AZhiKai** [comment]
Holding onto the merging for awhile. Figuring out what's the issue with `coveralls` not reporting coverage.

**Pull request merged**